mava/wrappers/gigastep.py

```diff
@@ -132,55 +118,57 @@ def reset(self, key: PRNGKey) -> Tuple[GigasteplState, TimeStep]:
+        state = GigastepState(state, key, 0, adversary_actions)
+        timestep = restart(obs, shape=(self.num_agents,), extras={"won_episode": False})
+        return state, timestep
 
-    def step(self, state: GigasteplState, action: Array) -> Tuple[GigasteplState, TimeStep]:
+    def step(self, state: GigastepState, action: Array) -> Tuple[GigastepState, TimeStep]:
         """
-        Takes a step in the Gigastepl environment.
+        Takes a step in the Gigastep environment.
 
         Args:
-            state (GigasteplState): The current state of the environment.
+            state (GigastepState): The current state of the environment.
             action (Array): The actions for controllable agents.
 
         Returns:
-            Tuple[GigasteplState, TimeStep]: A tuple containing the next state of the environment
+            Tuple[GigastepState, TimeStep]: A tuple containing the next state of the environment
             and the next time step.
 
         """
         key, step_key, adversary_key = jax.random.split(state.key, 3)
 
-        action = jnp.concatenate([action, state.Adverseary_action], axis=0, dtype=jnp.int16)
+        action = jnp.concatenate([action, state.adversary_action], axis=0, dtype=jnp.int16)
 
         obs, env_state, rewards, dones, ep_done = self._env.step(state.state, action, step_key)
 
+        # cut out the rewards,dones of the adversary
         rewards, dones = (
             rewards[: self.num_agents],
             dones[: self.num_agents],
-        )  # cut out the rewards,dones of the adversary
+        )
+
         team1_obs, team1_state, team2_obs, team2_state = self._split_obs_and_state(obs, env_state)
-        adversary_actions = self.adversary_policy(
-            team2_obs, team2_state, adversary_key
-        )  # take the actions of the adversary and cache it befor returning the new state
+
+        # take the actions of the adversary and cache it before returning the new state
+        adversary_actions = self.adversary_policy(team2_obs, team2_state, adversary_key)
 
         if self.has_global_state:
             obs = ObservationGlobalState(
                 agents_view=team1_obs,
                 action_mask=self.action_mask(),
                 global_state=self.get_global_state(obs, adversary_actions),
-                step_count=jnp.zeros(self.num_agents, dtype=int),
+                step_count=jnp.repeat(state.step, self.num_agents),
             )
         else:
             obs = Observation(
                 agents_view=team1_obs,
                 action_mask=self.action_mask(),
-                step_count=jnp.zeros(self.num_agents, dtype=int),
+                step_count=jnp.repeat(state.step, self.num_agents),
             )
 
         step_type = jax.lax.select(ep_done, StepType.LAST, StepType.MID)
 
-        current_winner = jnp.where(ep_done, self.get_wining_team(env_state), jnp.nan)
+        current_winner = jax.lax.select(ep_done, self.get_wining_team(env_state), False)
```

Comment:
Same here for avoiding select, please double check my logic:

```suggestion
        current_winner = ep_done & self.get_wining_team(env_state)
```

Reply:
Also can we change the method and variable name to `won_episode` to keep it inline with the extras